pkg/cmd/experimental/syncgroups/cli/prunegroups.go

```diff
@@ -0,0 +1,233 @@
+	// TODO(deads): enable this once we're able to support string slice elements that have commas
+	// cmd.Flags().StringSliceVar(&options.Blacklist, "blacklist-group", options.Blacklist, "group to blacklist")
+	cmd.Flags().StringVar(&configFile, "sync-config", configFile, "path to the sync config")
+	cmd.Flags().BoolVar(&options.Confirm, "confirm", false, "if true, modify OpenShift groups; if false, display groups")
+
+	return cmd
+}
+
+func (o *PruneGroupsOptions) Complete(whitelistFile, blacklistFile, configFile string, args []string, f *clientcmd.Factory) error {
+	var err error
+	o.Whitelist, err = buildOpenShiftGroupNameList(args, whitelistFile)
+	if err != nil {
+		return err
+	}
+
+	o.Blacklist, err = buildOpenShiftGroupNameList([]string{}, blacklistFile)
+	if err != nil {
+		return err
+	}
+
+	o.Config, err = decodeSyncConfigFromFile(configFile)
+	if err != nil {
+		return err
+	}
+
+	osClient, _, err := f.Clients()
+	if err != nil {
+		return err
+	}
+	o.GroupInterface = osClient.Groups()
+
+	return nil
+}
+
+func (o *PruneGroupsOptions) Validate() error {
+	results := validation.ValidateLDAPSyncConfig(o.Config)
+	if o.GroupInterface == nil {
+		results.Errors = append(results.Errors, fmt.Errorf("an OpenShift group client is required"))
+	}
+	// TODO(skuznets): pretty-print validation results
+	if len(results.Errors) > 0 {
+		return fmt.Errorf("validation of LDAP sync config failed: %v", kerrs.NewAggregate([]error(results.Errors)))
+	}
+	return nil
+}
+
+// Run creates the GroupSyncer specified and runs it to sync groups
+// the arguments are only here because its the only way to get the printer we need
+func (o *PruneGroupsOptions) Run(cmd *cobra.Command, f *clientcmd.Factory) error {
+	clientConfig, err := ldaputil.NewLDAPClientConfig(o.Config.URL, o.Config.BindDN, o.Config.BindPassword, o.Config.CA, o.Config.Insecure)
+	if err != nil {
+		return fmt.Errorf("could not determine LDAP client configuration: %v", err)
+	}
+
+	pruneBuilder, err := buildPruneBuilder(clientConfig, o.Config.RFC2307Config, o.Config.ActiveDirectoryConfig, o.Config.AugmentedActiveDirectoryConfig)
+	if err != nil {
+		return err
+	}
+
+	// populate schema-independent pruner fields
+	pruner := &syncgroups.LDAPGroupPruner{
+		Host:        clientConfig.Host(),
+		GroupClient: o.GroupInterface,
+		DryRun:      !o.Confirm,
+
+		Out: o.Out,
+		Err: os.Stderr,
+	}
+
+	listerMapper, err := getOpenShiftGroupListerMapper(clientConfig.Host(), o)
+	if err != nil {
+		return err
+	}
+	pruner.GroupLister = listerMapper
+	pruner.GroupNameMapper = listerMapper
+
+	pruner.GroupDetector, err = pruneBuilder.GetGroupDetector()
+	if err != nil {
+		return err
+	}
+
+	// Now we run the pruner and report any errors
+	pruneErrors := pruner.Prune()
+	return kerrs.NewAggregate(pruneErrors)
+
+}
+
+func buildPruneBuilder(clientConfig ldapclient.Config, rfc2307 *api.RFC2307Config, ad *api.ActiveDirectoryConfig, augmentedAD *api.AugmentedActiveDirectoryConfig) (PruneBuilder, error) {
```

Comment:
I'll add that change to the next pull in this sequence, it already contains misc. refactors